### DISCUSSION TITLE
What is metadata_serialized & code??

### DISCUSSION THREAD

**Discussion by ajaythxkur** [aptos-labs/aptos-developer-discussions#518] [Questions] [answered]
`public entry fun publish_package_txn(owner: &signer, metadata_serialized: vector<u8>, code: vector<vector<u8>>)`
 how to publish a contract calling other contract

**Comment by gregnazario** [comment, marked as answer]
### How to get metadata_serialized and code?

There are more complicated ways to get it, but the easiest way is to give all the same compiler publish inputs and instead use build-publish-payload to give you the info
```
aptos move build-publish-payload
```
